am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetRolePermissionsIntegrationTest.java

```diff
@@ -106,6 +110,47 @@ void whenInsufficientRolePermissionsShouldReturnNull() {
         assertThat(rolePermissions).isNull();
     }
 
+    @Test
+    void whenRolePermissionIsHighestThenShowAllAttribute() {
+
+        Map.Entry<Set<Permission>, SecurityClassification> readPermission = new Pair<>(ImmutableSet.of(READ), PUBLIC);
+        Map.Entry<Set<Permission>, SecurityClassification> readPermissionPrivate =
+            new Pair<>(ImmutableSet.of(READ), PRIVATE);
+
+        ImmutableSet multiPermission = ImmutableSet.builder().add(READ).add(CREATE).build();
+
+        Map.Entry<Set<Permission>, SecurityClassification> readPermissionRestricted =
+            new Pair(multiPermission, RESTRICTED);
+
+        importerService.addRole(roleName, IDAM, RESTRICTED, ROLE_BASED);
+
+        Map attributePermissionsForRole = ImmutableMap.of(
+            JsonPointer.valueOf(""), readPermission,
+            JsonPointer.valueOf("/address"), readPermissionPrivate,
+            JsonPointer.valueOf("/address/line1"), readPermissionRestricted
+        );
+
+        importerService.grantDefaultPermission(DefaultPermissionGrant.builder()
+            .roleName(roleName)
+            .resourceDefinition(ResourceDefinition.builder()
+                .serviceName(serviceName)
+                .resourceType(RESOURCE_TYPE)
+                .resourceName(RESOURCE_NAME)
+                .build())
+            .attributePermissions(attributePermissionsForRole)
+            .build());
+
+        RolePermissions rolePermissions =
+            service.getRolePermissions(buildResource(serviceName, RESOURCE_TYPE, RESOURCE_NAME), roleName);
+
+
+        assertThat(rolePermissions.getPermissions())
+            .hasSize(3)
+            .contains(new AbstractMap.SimpleEntry<>(JsonPointer.valueOf("/address"), ImmutableSet.of(READ)))
+            .containsEntry(JsonPointer.valueOf("/address/line1"), multiPermission)
+            .contains(new AbstractMap.SimpleEntry<>(JsonPointer.valueOf(""),ImmutableSet.of(READ)));
+    }
+
```

Comment:
A lot of this is out of date with the most recent commit to AM-271-security-classifications-for-resource-attribute